pkg/scheduler/reconciliation/reconciliation_test.go

```diff
@@ -729,3 +729,93 @@ func dbConnection(t *testing.T) db.Connection {
 	}
 	return dbConn
 }
+
+func TestReconciliationParallel(t *testing.T) {
+
+	t.Run("Create multiple instances of single reconciliations in parallel", func(t *testing.T) {
+		repo := newPersistentRepository(t)
+
+		inventory, err := cluster.NewInventory(dbConnection(t), true, cluster.MetricsCollectorMock{})
+		require.NoError(t, err)
+
+		errChannel := make(chan error, 100)
+		mockClusterState, _ := createClusterStates(t, inventory)
+
+		defer func() {
+			require.NoError(t, inventory.Delete(mockClusterState.Cluster.RuntimeID))
+		}()
+
+		startAt := time.Now().Add(1 * time.Second)
+		for i := 0; i < 50; i++ {
+			go func() {
+				time.Sleep(time.Until(startAt))
```

Comment:
If I understood correctly, you want to prevent the follow up code running, and wait for the next thread started. But `Sleep` will happen at each round within this for loop, which means it pause 3 time.

In single processors, there is no way you can make sure the created multiple threads started at the exactly same time, it's not how CPU works, CPU running sequentially, thread just win some CPU time slice when it's run, 
like this: https://en.wikipedia.org/wiki/Multithreading_(computer_architecture)#/media/File:Multithreaded_process.svg

If you really want to achieve parallel, since now days we all using multi core CPU, probably with some multiprocessors programming technique, you can configure thread running at each core at same time, but it's not controlled by our programmer with go code in the application level, it's down to OS level already.
https://stackoverflow.com/questions/20993139/how-does-goroutines-behave-on-a-multi-core-processor